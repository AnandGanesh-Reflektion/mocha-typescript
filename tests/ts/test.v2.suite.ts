import { suite, test, slow, timeout, skip, only, trait, skipOnError } from "../../index";

declare var describe, it;

suite("vanila tdd suite", () => {
    test("vanila test", () => {});
    test("vanila failing test", () => { throw new Error("x") });
    test("vanila asinc", done => setTimeout(done, 5));
    test("Vanila promise", () => new Promise((resolve, reject) => setTimeout(resolve, 5)));
    test.pending("pending", () => {});
    test.skip("skipped", () => {});
});

describe("vanila bdd suite", () => {
    it("vanila test", () => {});
    it("vanila failing test", () => { throw new Error("x") });
    it("vanila asinc", done => setTimeout(done, 5));
    it("Vanila promise", () => new Promise((resolve, reject) => setTimeout(resolve, 5)));
    it.skip("skipped", () => {});
});

@suite class Simple {
    @test test() {}
}

@suite(timeout(10))
class Timeouts {
    @test pass1(done) {
        setTimeout(done, 1);
    }
    @test pass2(done) {
        setTimeout(done, 20);
    }
    @test pass3(done) {
        setTimeout(done, 1);
    }
}

@suite(trait(ctx => ctx.timeout(10)))
class InlineTrait {
    @test timeout(done) {}
}

@suite(skipOnError)
class StockSequence {
    @test step1() {}
    @test step2() { throw new Error("Failed"); }
    @test step3() { /* should be skipped */ }
    @test step4() { /* should be skipped */ }
}

declare var beforeEach, afterEach;
var customSkipOnError = trait(function(ctx, ctor) {
    beforeEach(function() {
        if (ctor.__skip_all) {
            this.skip();
        }
    });
    afterEach(function() {
        if (this.currentTest.state === "failed") {
            ctor.__skip_all = true;
        }
    });
});

@suite(customSkipOnError)
class CustomSequence {
    @test step1() {}
    @test step2() { throw new Error("Failed"); }
    @test step3() { /* should be skipped */ }
    @test step4() { /* should be skipped */ }
}

@suite("name and trait", customSkipOnError)
class CustomSequence2 {
    @test step1() {}
    @test step2() { throw new Error("Failed"); }
    @test step3() { /* should be skipped */ }
}

@suite(slow(10))
class Slows {
    @test pass1(done) {
        setTimeout(done, 1);
    }
    @test pass2(done) {
        setTimeout(done, 20);
    }
    @test pass3(done) {
        setTimeout(done, 1);
    }
}

@suite("mocha typescript")
class Basic {
    @test "assert pass"() {
    }

    @test() "assert pass 2"() {
    }

    @test "test fail"() {
        throw new Error("Fail intentionally!");
    }

    @test() "test fail 2"() {
        throw new Error("Fail intentionally!");
    }
    
    @test.skip "test skip"() {
    }

    @test(timeout(5))
    "test intentinally timeout"(done) {
        setTimeout(done, 10);
    }

    @test(timeout(20))
    "test intentinall fail due error before timeout"(done) {
        setTimeout(() => done("Ooopsss..."), 5);
    }

    @test(timeout(100), slow(20))
    "test won't timeout but will be redish slow"(done) {
        setTimeout(done, 30);
    }

    @test(timeout(100), slow(20))
    "test won't timeout but will be yellowish slow"(done) {
        setTimeout(done, 15);
    }
}

@suite.skip
class Skipped1 {
    @test test() {}
}

@suite.skip()
class Skipped2 {
    @test test() {}
}

@suite.pending
class Pending1 {
    @test pending() {}
}

@suite.pending()
class Pending2 {
    @test pending() {}
}

@suite.pending(skipOnError)
class Pending3 {
    @test pending() {}
}

@suite("custom suite name")
class Test {
    @test("custom test name") test() {}
}