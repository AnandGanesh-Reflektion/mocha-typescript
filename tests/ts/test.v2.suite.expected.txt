

  vanila tdd suite
    ✓ vanila test
    1) vanila failing test
    ✓ vanila asinc
    ✓ Vanila promise
    - pending
    - skipped

  vanila bdd suite
    ✓ vanila test
    2) vanila failing test
    ✓ vanila asinc
    ✓ Vanila promise
    - skipped

  Simple
    ✓ test

  Timeouts
    ✓ pass1
    3) pass2
    ✓ pass3

  InlineTrait
    4) timeout

  StockSequence
    ✓ step1
    5) step2
    - step3
    - step4

  CustomSequence
    ✓ step1
    6) step2
    - step3
    - step4

  name and trait
    ✓ step1
    7) step2
    - step3

  Slows
    ✓ pass1
    ✓ pass2 (
    ✓ pass3

  mocha typescript
    ✓ assert pass
    ✓ assert pass 2
    8) test fail
    9) test fail 2
    - test skip
    10) test intentinally timeout
    11) test intentinall fail due error before timeout
    ✓ test won't timeout but will be redish slow (
    ✓ test won't timeout but will be yellowish slow (

  Skipped1
    - test

  Skipped2
    - test

  Pending1
    - pending

  Pending2
    - pending

  Pending3
    - pending

  custom suite name
    ✓ custom test name

  InstanceTests
    ✓ test
    12) testFailing
    ✓ beforeAfterCalled

  AsyncInstanceTests
    ✓ test
    13) testFailing
    ✓ beforeAfterCalled


  24 passing
  14 pending
  13 failing

